src/sentry/snuba/query_subscription_consumer.py

```diff
@@ -46,22 +217,82 @@ class InvalidSchemaError(InvalidMessageError):
     pass
 
 
+class QuerySubscriptionStrategyFactory(ProcessingStrategyFactory[KafkaPayload]):
+    def __init__(self, topic: str):
+        self.topic = topic
+
+    def create_with_partitions(
+        self,
+        commit: Commit,
+        partitions: Mapping[Partition, int],
+    ) -> ProcessingStrategy[KafkaPayload]:
+        def process_message(message: Message[KafkaPayload]) -> None:
+            with sentry_sdk.start_transaction(
+                op="handle_message",
+                name="query_subscription_consumer_process_message",
+                sampled=random() <= options.get("subscriptions-query.sample-rate"),
+            ), metrics.timer("snuba_query_subscriber.handle_message"):
+                value = message.value
+                assert isinstance(value, BrokerValue)
+                offset = value.offset
+                partition = value.partition.index
+                message_value = value.payload.value
+                try:
+                    handle_message(
+                        message_value,
+                        offset,
+                        partition,
+                        self.topic,
+                    )
+                except Exception:
+                    # This is a failsafe to make sure that no individual message will block this
+                    # consumer. If we see errors occurring here they need to be investigated to
+                    # make sure that we're not dropping legitimate messages.
+                    logger.exception(
+                        "Unexpected error while handling message in QuerySubscriptionStrategy. Skipping message.",
+                        extra={
+                            "offset": offset,
+                            "partition": partition,
+                            "value": message_value,
+                        },
+                    )
+
+        return RunTask(process_message, CommitOffsets(commit))
+
+
+def get_query_subscription_consumer(
+    topic: str,
+    group_id: str,
+    strict_offset_reset: bool,
+    initial_offset_reset: str = "earliest",
```

Comment:
Please don't set this default here, force the caller to pass it. It's very unexpected that it's different from the default in the CLI which is latest.

Reply:
Oops will do, I was copying what the existing consumer had.